kategory-test/src/main/kotlin/kategory/laws/OptionalLaws.kt

```diff
@@ -0,0 +1,98 @@
+package kategory
+
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+import kategory.optics.Optional
+
+object OptionalLaws {
+
+    inline fun <reified A, reified B> laws(optional: Optional<A, B>, aGen: Gen<A>, bGen: Gen<B>, funcGen: Gen<(B) -> B>, EQA: Eq<A>, EQB: Eq<B>): List<Law> = listOf(
+            Law("Optional Law: set what you get", { getOptionSet(optional, aGen, EQA) }),
+            Law("Optional Law: get what you get", { getGetOption(optional, aGen, bGen, EQB) }),
+            Law("Optional Law: set is idempotent", { setIdempotent(optional, aGen, bGen, EQA) }),
+            Law("Optional Law: modify identity = identity", { modifyIdentity(optional, aGen, EQA) }),
+            Law("Optional Law: compose modify", { composeModify(optional, aGen, funcGen, EQA) }),
+            Law("Optional Law: consistent set with modify", { consistentSetModify(optional, aGen, bGen, EQA) }),
+            Law("Optional Law: consistent modify with modify identity", { consistentModifyModifyId(optional, aGen, funcGen, EQA) }),
+            Law("Optional Law: consistent getOption with modify identity", { consistentGetOptionModifyId(optional, aGen, EQB) })
+    )
+
+    fun <A, B> getOptionSet(optional: Optional<A, B>, aGen: Gen<A>, EQA: Eq<A>): Unit =
+            forAll(aGen, { a ->
+                EQA.eqv(
```

Comment:
Is it worth using equalsUnderTheLaw here? We can get consistent changes in the future for all laws, and works like this behind the scenes.

Reply:
Yes, that should be `equalsUnderTheLaw`